install.zasilkovna.php

```diff
@@ -130,63 +178,48 @@ public function postflight($route, JAdapterInstance $adapter) {
-							  `weight` decimal(10,4) DEFAULT NULL,
-							  `branch_id` decimal(10,0) DEFAULT NULL,
-							  `branch_currency` char(5) DEFAULT NULL,
-							  `branch_name_street` varchar(500) DEFAULT NULL,
-							  `is_carrier` smallint(1) NOT NULL DEFAULT '0',
-							  `carrier_pickup_point` varchar(40) DEFAULT NULL,
-							  `email` varchar(255) DEFAULT NULL,
-							  `phone` varchar(255) DEFAULT NULL,
-							  `first_name` varchar(255) DEFAULT NULL,
-							  `last_name` varchar(255) DEFAULT NULL,
-							  `address` varchar(255) DEFAULT NULL,
-							  `city` varchar(255) DEFAULT NULL,
-							  `zip_code` varchar(255) DEFAULT NULL,
-							  `virtuemart_country_id` varchar(255) DEFAULT NULL,
-							  `adult_content` smallint(1) DEFAULT '0',
-							  `is_cod` smallint(1) DEFAULT NULL,
-							  `packet_cod` decimal(15,2) DEFAULT '0',
-							  `exported` smallint(1) DEFAULT NULL,
-							  `printed_label` smallint(1) DEFAULT '0',
-							  `shipment_name` varchar(5000) DEFAULT NULL,
-							  `shipment_cost` decimal(10,2) DEFAULT NULL,
-							  `shipment_package_fee` decimal(10,2) DEFAULT NULL,
-							  `tax_id` smallint(1) DEFAULT NULL,
-							  `created_on` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-							  `created_by` int(11) NOT NULL DEFAULT '0',
-							  `modified_on` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-							  `modified_by` int(11) NOT NULL DEFAULT '0',
-							  `locked_on` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
-							  `locked_by` int(11) NOT NULL DEFAULT '0',
-							  PRIMARY KEY (`id`)
-							) ENGINE=MyISAM DEFAULT CHARSET=utf8 COMMENT='zasilkovna';";
-			$db->setQuery($q);
-			$db->query();
-
 			$q = "
 INSERT INTO #__virtuemart_adminmenuentries (`module_id`, `parent_id`, `name`, `link`, `depends`, `icon_class`, `ordering`, `published`, `tooltip`, `view`, `task`) VALUES
 							(5, 0, 'ZASILKOVNA', '', '', 'vmicon vmicon-16-zasilkovna', 1, 1, '', 'zasilkovna', '');";
 			$db->setQuery($q);
-			$db->query();
+			$db->execute();
 
 		}
+
+        if ($route === 'update' && $this->fromVersion && version_compare($this->fromVersion, '1.2.0') < 0) {
+            $this->migratePricingRules();
+        }
+	}
+
+    public function pluginTableExists() {
+        $db = JFactory::getDBO();
+        $db->setQuery('SHOW TABLES LIKE "%_virtuemart_shipment_plg_zasilkovna"');
+        $row = $db->loadColumn();
+        return !empty($row);
+    }
+
+    public function upgradeSchema() {
+        $oldColumns = [];
+        if ($this->pluginTableExists()) {
+            $db = JFactory::getDBO();
+            $db->setQuery('SHOW FULL COLUMNS FROM `#__virtuemart_shipment_plg_zasilkovna`');
+            $oldColumns = $db->loadColumn();
+        }
+
+        $updater = new GenericTableUpdater();
+        $updater->updateMyVmTables(__DIR__ . '/install.sql');
+        $updater->updateMyVmTables(__DIR__ . '/install.sql'); // in order to use InnoDB we need to call the method twice to invoke alter logic
```

Comment:
Prosím vysvětlit. Pokud tam tabulky nejsou, tak je založí, pokud jsou, tak provede tu "alter logic" ne?

Reply:
Když se tabulka vytvoří, tak se jí nastaví ENGINE MyISAM. To já nechci.

Reply:
v tom případě bych (na řádku před druhým voláním) napsal "tables are created with MyISAM engine, to use InnoDB, we call the method for second time"